src/SourceCode.Clay.Data/SqlClient/SqlConnectionRetryOptions.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Diagnostics.Contracts;
+
+namespace SourceCode.Clay.Data.SqlClient
+{
+    /// <summary>
+    /// A property bag to hold various connection retry options.
+    /// </summary>
+    public sealed class SqlConnectionRetryOptions
+    {
+        #region Constants
+
+        public static SqlConnectionRetryOptions Default { [Pure]get; } = new SqlConnectionRetryOptions();
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// The number of reconnections attempted after identifying that there was an
+        /// idle connection failure. This must be an integer between 0 and 255. Default is 5.
+        /// Set to 0 to disable reconnecting on idle connection failures.
+        /// </summary>
+        public byte ConnectRetryCount { [Pure]get; } = 5;
```

Comment:
No need for options types to be immutable. This improves usability:

`new SqlRetryOptions() {
   ConnectRetryInterval = 100
}`